packages/site/src/api.ts

```diff
@@ -0,0 +1,126 @@
+import { BaseQueryFn, createApi } from '@reduxjs/toolkit/query/react';
+import { MetaMaskInpageProvider } from '@metamask/providers';
+import { RequestArguments } from '@metamask/providers/dist/BaseProvider';
+import { JsonRpcError, JsonRpcParams } from '@metamask/utils';
+
+declare global {
+  interface Window {
+    ethereum: MetaMaskInpageProvider;
+  }
+}
+
+export enum Tag {
+  InstalledSnaps = 'Installed Snaps',
+  TestState = 'Test State',
+}
+
+/**
+ * Base request function for all API calls.
+ *
+ * @param args - The request arguments.
+ * @param args.method - The RPC method to call.
+ * @param args.params - The parameters to pass to the RPC method.
+ * @returns The response from the RPC method.
+ */
+export const request: BaseQueryFn<RequestArguments> = async ({
+  method,
+  params,
+}) => {
+  try {
+    const data = await window.ethereum.request({ method, params });
+
+    return { data };
+  } catch (error) {
+    console.error(error);
+    return { error };
+  }
+};
+
+export type GetSnapsResult = Record<
+  string,
+  { error?: string; version: string }
+>;
+
+export interface InvokeSnapArgs {
+  snapId: string;
+  method: string;
+  params?: JsonRpcParams;
+}
+
+export type InvokeSnapResult = unknown;
+
+export interface InstallSnapArgs {
+  snapId: string;
+  version: string;
+}
+
+export interface InstallSnapResult {
+  snaps: {
+    [key: string]: {
+      error: JsonRpcError;
+    };
+  };
+}
+
+export const baseApi = createApi({
+  reducerPath: 'base',
+  baseQuery: request,
+  tagTypes: [Tag.InstalledSnaps, Tag.TestState],
+  endpoints(build) {
+    return {
+      getSnaps: build.query<GetSnapsResult, void>({
+        query: () => ({
+          method: 'wallet_getSnaps',
+        }),
+        providesTags: [Tag.InstalledSnaps],
+      }),
+
+      getState: build.query<unknown, { snapId: string }>({
+        query: ({ snapId }) => ({
+          method: 'wallet_invokeSnap',
+          params: [snapId, { method: 'retrieveTestData' }],
```

Comment:
Should this be parameterized?

Reply:
What do you mean?

Reply:
To not always use that method for reusability